src/compute-client/src/controller/instance.rs

```diff
@@ -1023,20 +1009,21 @@ where
         otel_ctx: OpenTelemetryContext,
         replica_id: ReplicaId,
     ) -> Option<ComputeControllerResponse<T>> {
-        let peek = match self.compute.peeks.get_mut(&uuid) {
-            Some(peek) => peek,
-            None => {
-                tracing::warn!("did not find pending peek for {}", uuid);
-                return None;
-            }
-        };
-
         // Forward the peek response, if we didn't already forward a response
         // to this peek previously. If the peek is targeting a replica, only
         // forward the response from that replica.
         // TODO: we could collect the other responses to assert equivalence?
         // Trades resources (memory) for reassurances; idk which is best.
-        //
+
+        let peek = self.compute.peeks.get(&uuid)?;
+
+        let target_replica = peek.target_replica.unwrap_or(replica_id);
+        if target_replica != replica_id {
+            return None;
+        }
+
+        self.remove_peeks(&[uuid].into());
+
         // NOTE: we use the `otel_ctx` from the response, not the
```

Comment:
stray comment. But I'm glad  `otel_ctx` is not used anymore to signal if the peek has been served :-)

Reply:
Why do you think that is a stray comment? It's still true that we use the `otel_ctx` from the response, not the one we have stored in the pending peek.

Reply:
Ah sorry I was referring this part 2 lines below:
```
        // ... We
        // still `take` the pending peek's `otel_ctx` to mark it as
        // served.
```

Reply:
Ah, good catch, thanks!